README.md

```diff
@@ -11,9 +11,64 @@ A [MiniJava](https://pp.ipd.kit.edu/lehre/WS201819/compprakt/intern/sprachberich
 ```bash
 curl https://sh.rustup.rs -sSf | sh
 rustup toolchain install nightly-2018-10-14
+```
+
+## Tools
+
+We're using the tools [`rustfmt`](https://github.com/rust-lang-nursery/rustfmt)
+to format our code and
+[`clippy`](https://github.com/rust-lang-nursery/rust-clippy) as a linter to keep
+the code clean, idiomatic and correct. These tools are available in
+`nightly-2018-10-14` via `rustup`:
+
+```bash
 rustup component add clippy-preview
+rustup component add rustfmt-preview
+```
+
+To run those tools use
+
+```bash
+cargo clippy
+cargo fmt --all
+```
+
+For code that should **not** get formatted by `rustfmt` mark the code with
+
+```rust
+#[rustfmt::skip]
 ```
 
+A Clippy lint can be disabled similar to `rustc` lints:
+
+```rust
+#![allow(clippy::lint_group)]
+#[allow(clippy::lint_name)]
+```
+
+## Workflow
+
+To keep the code quality high and always have a master branch, that passes all
+test all the time ([The Not Rocket Science
+Rule](https://graydon2.dreamwidth.org/1597.html)), we aim to only add code to
+the project via Pull Requests. Every PR should get reviewed by another person to
+ensure keeping the code quality high (typos and simple one-line fixes can be
+accepted by the author). After that the reviewer should use `bors` to merge the
+PR. For big and/or major changes it is recommended to have at least two
+reviewers.
+
+Before creating a PR make sure to run
+
+```bash
+cargo clippy
```

Comment:
`./build --ci`?

Reply:
Depends on #5. Currently rustfmt is only run with `--check` in the `build` script, which won't format the code.